pinot-core/src/main/java/org/apache/pinot/core/operator/filter/H3InclusionIndexFilterOperator.java

```diff
@@ -70,10 +70,10 @@ public H3InclusionIndexFilterOperator(IndexSegment segment, QueryContext queryCo
     _isPositiveCheck = BooleanUtils.toBoolean(eqPredicate.getValue());
 
     if (arguments.get(0).getType() == ExpressionContext.Type.IDENTIFIER) {
-      _h3IndexReader = segment.getDataSource(arguments.get(0).getIdentifier()).getH3Index();
+      _h3IndexReader = segment.getDataSource(arguments.get(0).getIdentifier(), queryContext.getSchema()).getH3Index();
```

Comment:
This one shouldn't try to use virtual data source. Same for some other invocations. Please double check all the usages